packages/superset-ui-dimension/src/getTextDimension.ts

```diff
@@ -19,13 +19,27 @@ export interface GetTextDimensionInput {
 
 const DEFAULT_DIMENSION = { height: 20, width: 100 };
 
+export function createSVGNode(container: HTMLElement): SVGSVGElement {
+  const textNode = document.createElementNS(SVG_NS, 'text');
+  const svg = document.createElementNS(SVG_NS, 'svg');
+  svg.style.position = 'absolute'; // so it won't disrupt page layout
+  svg.style.opacity = '0'; // and not visible
+  svg.style.pointerEvents = 'none'; // and not capturing mouse events
+  svg.appendChild(textNode);
+  container.appendChild(svg);
+
+  return svg;
+}
+
 export default function getTextDimension(
   input: GetTextDimensionInput,
   defaultDimension: Dimension = DEFAULT_DIMENSION,
+  existingSVGNode?: SVGSVGElement,
 ): Dimension {
   const { text, className, style = {}, container = document.body } = input;
 
-  const textNode = document.createElementNS(SVG_NS, 'text');
+  const svg = existingSVGNode || createSVGNode(container);
+  const textNode = svg.lastElementChild as SVGSVGElement;
```

Comment:
You will have to remove `className` and remove all styles from this text node before continuing. I am not sure how much performance gain do you get vs. creating a new text node. The `svg` node seems easily reusable.

Reply:
I'll give it a try with creating a new one.

By reusing the `svg` node, the time spent in `getTextDimension` went down from 20 seconds to less than a second for a very large table. I'll benchmark to get the exact numbers.

Reply:
You could also try to hash `className` and `textStyle` and store hash in the text element. If the hash is the same, then no need to recreate `<text>`.